server/sonar-server/src/test/java/org/sonar/server/computation/analysis/AnalysisMetadataHolderRule.java

```diff
@@ -108,4 +112,15 @@ public int getRootComponentRef() {
     checkState(rootComponentRef.isInitialized(), "Root component ref has not been set");
     return rootComponentRef.getProperty();
   }
+
+  public AnalysisMetadataHolderRule setQProfilesByLanguage(Map<String, QualityProfile> qProfilesPerLanguage) {
+    this.qProfilesPerLanguage.setProperty(qProfilesPerLanguage);
```

Comment:
checkState(!this.qProfilesPerLanguage.isInitialized(), "QProfiles by language has already been set") is missing